core/vm/lfvm/interpreter_test.go

```diff
@@ -18,6 +21,392 @@ type example struct {
+	stackPtrPos int
+	argData     uint16
+	endStatus   Status
+	isBerlin    bool
+	isLondon    bool
+	mockCalls   func(mockStateDB *vm.MockStateDB)
+	gasStart    uint64
+	gasConsumed uint64
+}
+
+func getInstructions(start OpCode, end OpCode) (opCodes []OpCode) {
+	for i := start; i <= end; i++ {
+		opCodes = append(opCodes, OpCode(i))
+	}
+	return
+}
+
+func getInstructionsWithGas(start OpCode, end OpCode, gas uint64) (opCodes []OpCodeWithGas) {
+	for i := start; i <= end; i++ {
+		opCode := OpCodeWithGas{OpCode(i), gas}
+		opCodes = append(opCodes, opCode)
+	}
+	return
+}
+
+var fullStackFailOpCodes = []OpCode{
+	MSIZE, ADDRESS, ORIGIN, CALLER, CALLVALUE, CALLDATASIZE,
+	CODESIZE, GASPRICE, COINBASE, TIMESTAMP, NUMBER,
+	DIFFICULTY, GASLIMIT, PC, GAS, RETURNDATASIZE,
+	SELFBALANCE, CHAINID, BASEFEE,
+	PUSH1_PUSH1_PUSH1_SHL_SUB,
+	PUSH1_DUP1, PUSH1_PUSH1, PUSH1_PUSH4_DUP3,
+}
+
+var emptyStackFailOpCodes = []OpCode{
+	POP, ADD, SUB, MUL, DIV, SDIV, MOD, SMOD, EXP, SIGNEXTEND,
+	SHA3, LT, GT, SLT, SGT, EQ, AND, XOR, OR, BYTE,
+	SHL, SHR, SAR, ADDMOD, MULMOD, ISZERO, NOT, BALANCE, CALLDATALOAD, EXTCODESIZE,
+	BLOCKHASH, MLOAD, SLOAD, EXTCODEHASH, JUMP, SELFDESTRUCT,
+	MSTORE, MSTORE8, SSTORE, JUMPI, RETURN, REVERT,
+	CALLDATACOPY, CODECOPY, RETURNDATACOPY,
+	EXTCODECOPY, CREATE, CREATE2, CALL, CALLCODE,
+	STATICCALL, DELEGATECALL, POP_POP, POP_JUMP,
+	SWAP2_POP, PUSH1_ADD, PUSH1_SHL, SWAP2_SWAP1_POP_JUMP,
+	PUSH2_JUMPI, ISZERO_PUSH2_JUMPI, SWAP2_SWAP1,
+	DUP2_LT, SWAP1_POP_SWAP2_SWAP1, POP_SWAP2_SWAP1_POP,
+	AND_SWAP1_POP_SWAP2_SWAP1, SWAP1_POP, DUP2_MSTORE,
+}
+
+func addFullStackFailOpCodes(tests []OpcodeTest) []OpcodeTest {
+	var addedTests []OpcodeTest
+	addedTests = append(addedTests, tests...)
+	var opCodes []OpCode
+	opCodes = append(opCodes, fullStackFailOpCodes...)
+	opCodes = append(opCodes, getInstructions(PUSH1, PUSH32)...)
+	opCodes = append(opCodes, getInstructions(DUP1, DUP16)...)
+	for _, opCode := range opCodes {
+		addedTests = append(addedTests, OpcodeTest{opCode.String() + " Stack overflow", []Instruction{{opCode, 1}}, MAX_STACK_SIZE, 0, ERROR, false, false, nil, GAS_START, 0})
+	}
+	return addedTests
+}
+
+func addEmptyStackFailOpCodes(tests []OpcodeTest) []OpcodeTest {
+	var addedTests []OpcodeTest
+	addedTests = append(addedTests, tests...)
+	var opCodes []OpCode
+	opCodes = append(opCodes, emptyStackFailOpCodes...)
+	opCodes = append(opCodes, getInstructions(DUP1, DUP16)...)
+	opCodes = append(opCodes, getInstructions(SWAP1, SWAP16)...)
+	opCodes = append(opCodes, getInstructions(LOG0, LOG4)...)
+	for _, opCode := range opCodes {
+		addedTests = append(addedTests, OpcodeTest{opCode.String() + " Stack underflow", []Instruction{{opCode, 1}}, 0, 0, ERROR, false, false, nil, GAS_START, 0})
+	}
+	return addedTests
+}
+
+func TestStackBoundry(t *testing.T) {
+
+	// Add tests for execution
+	tests := addFullStackFailOpCodes([]OpcodeTest{})
+	tests = addEmptyStackFailOpCodes(tests)
+
+	for _, test := range tests {
```

Comment:
Each case could be exposed to the testing framework as a sub-test using `t.Run(..)` [1]. This way, the name of the failing test will already give information about what went wrong.

[1] https://go.dev/blog/subtests

Reply:
Separated these two test cases, but I'll refactor it to sub-tests next time.